### PR TITLE
feat: denort binary for `deno compile`

### PR DESCRIPTION
**Pull request by littledivy**
This introduces the `denort` binary - a slim version of deno without tooling. The binary is used as the default for `deno compile`.

Improves `deno compile` final size by ~2.5x (141 MB -> 61 MB) on Linux x86_64.

### CODE REVIEW

File: cli/Cargo.toml
Comment (lucacasonato) [inline cli/Cargo.toml:20]: `denort` maybe? That's what it used to be called.

File: cli/main_standalone.rs
Comment (bartlomieju) [inline cli/main_standalone.rs:48]: This should cause an error

File: cli/worker.rs
Comment (mmastrac) [inline cli/worker.rs:83]: I think this would be cleaner if we provided a factory for HmrRunner and CoverageCollector instead of having a "half configured" pre-setup state. We'll probably get bitten by spurious `unwrap()` panicks down the road unless we enforce this via the type system.

ie:

`Box<dyn FnOnce(LocalInspectorSession)> -> Box<dyn CoverageCollector>`

**Review by mmastrac — COMMENTED** [review]
Looks good, but one small change requested before we land.

File: cli/Cargo.toml
Comment (ry) [reply cli/Cargo.toml]: maybe call it "denoy" ? i guess this name is never exposed to the user, so maybe it should favor clarity over brevity, but "denort" sounds like "deno real time" to me.

File: cli/worker.rs
Comment (littledivy) [reply cli/worker.rs]: Good point. I've added `create_hmr_runner` and `create_coverage_collector` creation callbacks.

File: cli/Cargo.toml
Comment (littledivy) [reply cli/Cargo.toml]: "denoy" is probably even more confusing 😅 

we've already known this as "denort" in the past, it makes sense to just continue using that term internally.

**Review by mmastrac — APPROVED** [review]
LGTM pending the name. In phase 2 we should hook up some basic smoke integration tests.

**Pull request merged**
